linera-base/src/identifiers.rs

```diff
@@ -318,6 +328,52 @@ pub struct StreamId {
     pub stream_name: StreamName,
 }
 
+impl Display for StreamId {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{}:{}",
+            self.application_id,
+            STANDARD_NO_PAD.encode(&self.stream_name.0)
+        )
+    }
+}
+
+/// An event identifier.
+#[derive(
+    Debug,
+    PartialEq,
+    Eq,
+    Hash,
+    Clone,
+    Serialize,
+    Deserialize,
+    WitLoad,
+    WitStore,
+    WitType,
+    SimpleObject,
+)]
+pub struct EventId {
+    /// The ID of the chain that generated this event.
+    pub chain_id: ChainId,
+    /// The ID of the stream this event belongs to.
+    pub stream_id: StreamId,
+    /// The event key.
+    pub key: Vec<u8>,
```

Comment:
`StreamId` is a tuple `ApplicationId` + `StreamName`. Is there anything in particular that you expect the `StreamName` could be? 

I'd imagine that `StreamName` is like a `topic` for messages sent over that stream. In that sense it's already _keyed_.

Reply:
Yes, I agree, it's like a topic, and there are no specific expectations: Every app can use it as needed.

Reply:
There was a hidden question there – maybe we can remove `key` if `StreamId` already has `StreamName` in it and there can be multiple streams created for the same app.

Or, alternatively, remove `StreamName` and just use the `key` here. GraphQL client can subscribe to a document where `key` is something specific.

Reply:
Events are emitted in a sequence anyway since contract execution is sequential and deterministic. The event id will be sth like `<chain_id>-<app_id>-<txn_id>-<seq_num>`

Reply:
Yes, the thought was that if we also have an index per _stream name_ (without the chain ID), i.e. kind of a stream that multiple chains can write to, it could be useful to use different keys. (The "latest" event in that multi-chain stream would be subjective then, of course.)